File: flutter/test/push_target_test.dart
import 'package:cyaim_im/cyaim_im.dart';
import 'package:test/test.dart';

/// `evt.call` and `evt.desk` were absent from every client SDK while the server was already
/// pushing them (AUDIT-2026-08-21 §6.1): the frame arrived, no constant matched it, and the app
/// dropped it without a word.
///
/// `evt.call` no longer exists on either side — calling was withdrawn from every product surface
/// on 2026-09-14 — so `evt.desk` is the only half left to pin.
///
/// The exhaustive cross-SDK check is `PushTargetParityTests` in `tests/IM.Tests.Unit` — it is the
/// only one that can read the server's own declaration. Dart cannot enumerate static consts
/// without mirrors, so an exhaustive list here would just be a second hand-written copy; this pins
/// the one that drifted and stayed.
void main() {
  test('desk is a named constant, not a bare string', () {
    expect(ImPushTarget.desk, 'evt.desk');
  });
}

File: typescript/test/push-target.test.ts
import assert from 'node:assert/strict';
import { describe, it } from 'node:test';

import { PushTarget, type PushTargetName } from '../src/protocol.js';

/**
 * `evt.call` and `evt.desk` shipped in the server and in SPEC-02 §2.7 while no client SDK named
 * them (AUDIT-2026-08-21 §6.1) — the frames arrived, nothing matched them, and applications
 * dropped them in silence.
 *
 * `evt.call` is gone from both sides as of 2026-09-14: the server withdrew calling from every
 * product surface, so `PushTarget` no longer declares it and this list no longer expects it. The
 * audit is still why this file exists — `evt.desk` is the half that survived, and it is pinned
 * below on its own. A reader who finds only one of the two names here should read this paragraph
 * rather than assume the other was dropped by accident.
 *
 * The authoritative guard is `PushTargetParityTests` in `tests/IM.Tests.Unit`, which reads all five
 * SDK tables and compares them against `PushTarget` in `ServerPush.cs`; only that one can see the
 * server. This suite exists so the omission is also visible to someone running `npm test` without a
 * .NET SDK on the machine.
 */
describe('push targets', () => {
  it('lists exactly the targets the server can push', () => {
    assert.deepEqual([...Object.values(PushTarget)].sort(), [
      'conn.kick', 'evt.conversationUpdate', 'evt.desk', 'evt.friend',
      'evt.group', 'evt.message', 'evt.messageUpdate', 'evt.presence', 'evt.read',
      'evt.stream', 'evt.system', 'evt.typing',
    ]);
  });

  it('keeps desk inside the union type', () => {
    const desk: PushTargetName = PushTarget.Desk;
    assert.equal(desk, 'evt.desk');
  });
});
